modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/AddEndPointSecurityPerTypeTestCase.java

```diff
@@ -0,0 +1,475 @@
+        String backendToken = authorization.replaceFirst("Bearer ", "");
+        validateIntrospectionResponse(user, backendToken, applicationKeyBeanProduction.getConsumerKey());
+        String sandAppTokenJti = TokenUtils.getJtiOfJwtToken(sandboxApplication.getToken().getAccessToken());
+        requestHeadersGet.put("Authorization", "Bearer " + sandAppTokenJti);
+        HttpResponse sandboxResponse =
+                HTTPSClientUtils.doGet(getAPIInvocationURLHttp(apiContext, API_VERSION_1_0_0), requestHeadersGet);
+        Assert.assertEquals(sandboxResponse.getResponseCode(), 200);
+        headers = sandboxResponse.getHeaders();
+        authorization = headers.get("BACKEND_AUTHORIZATION_HEADER");
+        Assert.assertNotNull(authorization);
+        Assert.assertTrue(authorization.contains("Bearer"));
+        backendToken = authorization.replaceFirst("Bearer ", "");
+        validateIntrospectionResponse(user, backendToken, applicationKeyBeanSandbox.getConsumerKey());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void cleanUpArtifacts() throws Exception {
+
+        restAPIStore.removeApplicationById(applicationID);
+        for (String apiId: apiIds) {
+            undeployAndDeleteAPIRevisionsUsingRest(apiId, restAPIPublisher);
+            restAPIPublisher.deleteAPI(apiId);
+        }
+        super.cleanUp();
+    }
+
+    private void validateIntrospectionResponse(User user, String accessToken, String clientId) throws UnsupportedEncodingException {
+
+        String introspectionUrl = "https://localhost:9943/oauth2/introspect";
+        if (!"carbon.super".equals(user.getUserDomain())) {
+            introspectionUrl = "https://localhost:9943/t/" + user.getUserDomain() + "/oauth2/introspect";
+        }
+        CloseableHttpClient closeableHttpClient = HttpClientBuilder.create().build();
+        HttpPost httpPost = new HttpPost();
+        httpPost.addHeader("Authorization",
+                "Basic " + Base64.encode(user.getUserName().concat(":").concat(user.getPassword()).getBytes()));
+        httpPost.setURI(URI.create(introspectionUrl));
+        UrlEncodedFormEntity urlEncodedFormEntity = new UrlEncodedFormEntity(Arrays.asList(new BasicNameValuePair(
+                "token", accessToken)));
+        httpPost.setEntity(urlEncodedFormEntity);
+        try (CloseableHttpResponse response = closeableHttpClient.execute(httpPost)) {
+            Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
+            HttpEntity entity = response.getEntity();
+            JSONObject jsonPayload = (JSONObject) new JSONParser().parse(new InputStreamReader(entity.getContent()));
+            Assert.assertTrue((Boolean) jsonPayload.get("active"));
+            Assert.assertNotNull(jsonPayload.get("client_id"));
+            Assert.assertEquals(jsonPayload.get("client_id"), clientId);
+        } catch (IOException | ParseException e) {
+            e.printStackTrace();
```

Comment:
Here instead of printing the entire stack trace, shall we catch ad throw the exception and print only the message here?

Reply:
Fixed via https://github.com/wso2/product-apim/pull/11444/commits/eb2a3225ca45b153d419d20862f9affb6874b92a